From 0f4023d5f43c520ac2c2e1a893b1ae0b1575f184 Mon Sep 17 00:00:00 2001
From: Luiz Capitulino <lcapitulino@redhat.com>
Date: Mon, 29 Mar 2010 16:14:43 -0300
Subject: [PATCH 08/23] Monitor: Convert do_closefd() to cmd_new_ret()

RH-Author: Luiz Capitulino <lcapitulino@redhat.com>
Message-id: <1269879298-30711-9-git-send-email-lcapitulino@redhat.com>
Patchwork-id: 8172
O-Subject: [PATCH 08/23] Monitor: Convert do_closefd() to cmd_new_ret()
Bugzilla: 563491
RH-Acked-by: Markus Armbruster <armbru@redhat.com>
RH-Acked-by: Juan Quintela <quintela@redhat.com>
RH-Acked-by: Kevin Wolf <kwolf@redhat.com>

Signed-off-by: Luiz Capitulino <lcapitulino@redhat.com>
Signed-off-by: Anthony Liguori <aliguori@us.ibm.com>
(cherry picked from commit aeb91c1e13d574ce71a408fb4cf836fbc45b40ab)
---
 monitor.c       |    5 +++--
 qemu-monitor.hx |    2 +-
 2 files changed, 4 insertions(+), 3 deletions(-)

Signed-off-by: Eduardo Habkost <ehabkost@redhat.com>
---
 monitor.c       |    5 +++--
 qemu-monitor.hx |    2 +-
 2 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/monitor.c b/monitor.c
index 0e65853..06caf03 100644
--- a/monitor.c
+++ b/monitor.c
@@ -2488,7 +2488,7 @@ static int do_getfd(Monitor *mon, const QDict *qdict, QObject **ret_data)
     return 0;
 }
 
-static void do_closefd(Monitor *mon, const QDict *qdict, QObject **ret_data)
+static int do_closefd(Monitor *mon, const QDict *qdict, QObject **ret_data)
 {
     const char *fdname = qdict_get_str(qdict, "fdname");
     mon_fd_t *monfd;
@@ -2502,10 +2502,11 @@ static void do_closefd(Monitor *mon, const QDict *qdict, QObject **ret_data)
         close(monfd->fd);
         qemu_free(monfd->name);
         qemu_free(monfd);
-        return;
+        return 0;
     }
 
     qemu_error_new(QERR_FD_NOT_FOUND, fdname);
+    return -1;
 }
 
 static void do_loadvm(Monitor *mon, const QDict *qdict)
diff --git a/qemu-monitor.hx b/qemu-monitor.hx
index 8ba2425..c22fee5 100644
--- a/qemu-monitor.hx
+++ b/qemu-monitor.hx
@@ -1042,7 +1042,7 @@ ETEXI
         .params     = "closefd name",
         .help       = "close a file descriptor previously passed via SCM rights",
         .user_print = monitor_user_noop,
-        .mhandler.cmd_new = do_closefd,
+        .cmd_new_ret = do_closefd,
     },
 
 STEXI
-- 
1.7.0.3

